zerver/tests/fixtures/markdown_test_cases.json

```diff
@@ -932,6 +932,111 @@
+    },
+    {
+      "name": "auto_shorten_issues_comment_links",
+      "input": "https://github.com/zulip/zulip/issues/16482#issuecomment-726354516",
+      "expected_output": "<p><a href=\"https://github.com/zulip/zulip/issues/16482#issuecomment-726354516\">zulip/zulip#16482#issuecomment-726354516</a></p>"
+    },
+    {
+      "name": "auto_shorten_files_links",
+      "input": "https://github.com/zulip/zulip/pull/16860/files",
+      "expected_output": "<p><a href=\"https://github.com/zulip/zulip/pull/16860/files\">zulip/zulip#16860/files</a></p>"
+    },
+    {
+      "name": "auto_shorten_files_comment_links",
+      "input": "https://github.com/zulip/zulip/pull/16860/files#r539133612",
+      "expected_output": "<p><a href=\"https://github.com/zulip/zulip/pull/16860/files#r539133612\">zulip/zulip#16860/files#r539133612</a></p>"
+    },
+    {
+      "name": "auto_shorten_commit_links",
+      "input": "https://github.com/zulip/zulip/commit",
+      "expected_output": "<p><a href=\"https://github.com/zulip/zulip/commit\">zulip/zulip/commit</a></p>"
+    },
+    {
+      "name": "auto_shorten_clone_links",
+      "input": "https://github.com/zulip/zulip.git",
+      "expected_output": "<p><a href=\"https://github.com/zulip/zulip.git\">zulip/zulip.git</a></p>"
+    },
+    {
+      "name": "auto_shorten_labels_links",
+      "input": "https://github.com/zulip/zulip/labels",
+      "expected_output": "<p><a href=\"https://github.com/zulip/zulip/labels\">zulip/zulip/labels</a></p>"
+    },
+    {
+      "name": "auto_shorten_labels_links",
+      "input": "https://github.com/zulip/zulip/labels/area%3A%20markdown",
+      "expected_output": "<p><a href=\"https://github.com/zulip/zulip/labels/area%3A%20markdown\">https://github.com/zulip/zulip/labels/area%3A%20markdown</a></p>"
+    },
+    {
+      "name": "auto_shorten_tree_links",
+      "input": "https://github.com/zulip/zulip/tree/chat.zulip.org",
+      "expected_output": "<p><a href=\"https://github.com/zulip/zulip/tree/chat.zulip.org\">https://github.com/zulip/zulip/tree/chat.zulip.org</a></p>"
+    },
+    {
+      "name": "auto_shorten_links",
+      "input": "https://github.com/foo/bar",
+      "expected_output": "<p><a href=\"https://github.com/foo/bar\">foo/bar</a></p>"
+    },
+    {
+      "name": "auto_shorten_links",
+      "input": "https://github.com/foo/bar/baz",
+      "expected_output": "<p><a href=\"https://github.com/foo/bar/baz\">foo/bar/baz</a></p>"
+    },
+    {
+      "name": "auto_shorten_links",
+      "input": "https://github.com/foo/bar/baz/zar",
+      "expected_output": "<p><a href=\"https://github.com/foo/bar/baz/zar\">https://github.com/foo/bar/baz/zar</a></p>"
+    },
+    {
+      "name": "auto_shorten_links",
+      "input": "https://github.com/foo",
+      "expected_output": "<p><a href=\"https://github.com/foo\">https://github.com/foo</a></p>"
```

Comment:
I would do specifically https://github.com/marketplace for the test here, since that's the sort of page we definitely don't want to turn into "marketplace".  I'm not entirely what a correct rule would be to exclude such pages, but one theory might be that we should only do this linkification when the `artifact` field is present?